File: apps/Image-to-3D/Android/app/src/main/java/ai/zetic/demo/imageto3d/AppConfig.kt
package ai.zetic.demo.imageto3d

/** Central configuration - mirrors iOS/ImageTo3D/App/AppConfig.swift. */
object AppConfig {
    // Melange model.
    const val PERSONAL_KEY = "YOUR_MLANGE_KEY"
    const val MODEL_NAME = "realtonypark/Depth-Anything-V2-Small"

    // Model I/O contract (fixed at export time; NPU shapes are hard-coded).
    // Input: (1, 3, 518, 518) float32, RGB in [0, 1]. ImageNet normalization
    // is baked into the exported graph. Output: (1, 518, 518) relative
    // inverse depth (larger = closer).
    const val INPUT_SIZE = 518

    // Depth to 3D reconstruction (planar relief: x/y stay on a fixed reference
    // plane so the photo keeps its shape; relative depth displaces z only).
    const val FOV_Y_DEGREES = 60f
    const val PLANE_Z = 2.0f
    const val RELIEF_DEPTH = 0.4f       // z displacement as a fraction of width
    const val MESH_STRIDE = 1           // full 518×518 vertex grid
    const val EDGE_THRESHOLD = 0.07f    // raw-disparity delta above which quads drop
}

File: apps/Image-to-3D/iOS/ImageTo3D/App/AppConfig.swift
import Foundation

/// Central configuration for the Image-to-3D pipeline.
enum AppConfig {
    // Melange model.
    static let personalKey = "YOUR_MLANGE_KEY"
    static let modelName = "realtonypark/Depth-Anything-V2-Small"
    static let modelVersion: Int? = nil   // nil = latest

    // Model I/O contract (fixed at export time; NPU shapes are hard-coded).
    // Input: (1, 3, 518, 518) float32, RGB in [0, 1]. ImageNet normalization
    // is baked into the exported graph. Output: (1, 518, 518) relative
    // inverse depth (larger = closer).
    static let inputSize = 518

    // Depth to 3D reconstruction (planar relief: x/y stay on a fixed reference
    // plane so the photo keeps its shape; relative depth displaces z only.
    // Mapping the full disparity range through a pinhole instead turns the
    // mesh into a deep frustum cone; background pixels blow up laterally).
    static let fovYDegrees: Float = 60          // display camera FOV
    static let planeZ: Float = 2.0              // reference plane distance
    static let reliefDepth: Float = 0.4         // z displacement as a fraction
                                                // of the photo's world width
    static let meshStride = 1                   // full 518x518 vertex grid;
                                                // quad-level silhouette steps
                                                // are invisible at this size
                                                // (mesh build ~15 ms in Release)
    static let edgeThreshold: Float = 0.07      // normalized-disparity delta above
                                                // which quads are dropped
}
